lib/rasr_cache.py

```diff
@@ -833,6 +833,58 @@ def read_v(self, size, a):
         a.fromfile(self.f, size)
         return a
 
+    # write routines
+    def write_str(self, s):
+        """
+        :param str s:
+        :rtype: int
+        """
+        return self.f.write(pack("%ds" % len(s), s.encode("ascii")))
```

Comment:
```suggestion
    def write_str(self, s, enc="ascii"):
        """
        :param str s:
        :rtype: int
        """
        return self.f.write(pack("%ds" % len(s), s.encode(enc)))
```
make encoding a parameter like in the read function.